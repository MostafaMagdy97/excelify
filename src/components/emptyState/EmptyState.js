import React from 'react';
import { Flex } from 'grid-styled';

import Text from '../text/Text';
import Icon from '../icon/Icon';
import IconsStore from '../icon/IconsStore';

import { FONT_TYPES } from '../../base/Typography';
import { COLORS } from '../../base/Colors';

type Props = {
  icon: string,
  text: string,
};

const EmptyState = (props: Props) => (
  <Flex alignItems="center" justifyContent="center" flexDirection="column" width={1}>
    {props.icon && (
      <Icon icon={IconsStore.getIcon(props.icon)} width={50} color={COLORS.DISABLED} />
    )}
    <Text type={FONT_TYPES.TITLE} color={COLORS.DISABLED} mt={3}>
      {props.text}
    </Text>
  </Flex>
);

export default EmptyState;
