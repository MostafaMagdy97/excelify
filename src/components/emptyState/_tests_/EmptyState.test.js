import React from 'react';
import { configure, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-enzyme';
import 'jest-styled-components';

import EmptyState from '../EmptyState';

configure({ adapter: new Adapter() });

describe('EmptyState', () => {
  it('should render', () => {
    const tree = render(<EmptyState />);
    expect(tree).toMatchSnapshot();
  });
});
