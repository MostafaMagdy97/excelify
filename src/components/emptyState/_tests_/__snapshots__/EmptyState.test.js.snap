// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState should render 1`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  margin-top: 16px;
  color: disabled;
  font-weight: 500;
  font-size: 21px;
  font-display: fallback;
}

@media screen and (min-width:40em) {
  .c1 {
    font-size: 21px;
  }
}

@media screen and (min-width:52em) {
  .c1 {
    font-size: 20px;
  }
}

@media screen and (min-width:64em) {
  .c1 {
    font-size: 20px;
  }
}

<div
  class="c0"
>
  <label
    class="c1"
    color="disabled"
    font-size="21px,21px,20px,20px"
    font-weight="500"
    mt="3"
    tag="label"
    type="title"
  >
    
  </label>
</div>
`;
